<?xml version="1.0"?>
<launch>

    <include file="$(find husky_gazebo)/launch/spawn_husky.launch">
        <arg name="x"                   value="2" />
        <arg name="y"                   value="2" />
    </include>

    <include file="$(find mav_linear_mpc)/launch/mav_linear_mpc_sim.launch" />

    <node pkg="auto_landing" type="auto_landing_detect_platform_node" name="auto_landing_detect_platform_node" output="screen">
    <rosparam file="$(find auto_landing)/cfg/detect_platform_params.yaml" />
    <remap from="image_raw"   to="/firefly/camera_front/image_raw" />
    <remap from="height_quad" to="/firefly/ground_truth/odometry" />
    </node>
    <arg name="trajectory" default="linear"/>
    <arg name="mav_name"             default="firefly"/>
    <arg name="world_name"           default="basic"/>
    <arg name="enable_logging"       default="false"/>
    <arg name="enable_ground_truth"  default="true"/>
    <arg name="log_file"             default="$(arg mav_name)"/>
    <arg name="debug"                default="false"/>
    <arg name="gui"                  default="true"/>
    <arg name="paused"               default="true"/>

    <!-- The following line causes gzmsg and gzerr messages to be printed to the console
        (even when Gazebo is started through roslaunch) -->
   <arg name="verbose"              default="false"/>
    <env name="GAZEBO_MODEL_PATH"       value="${GAZEBO_MODEL_PATH}:$(find rotors_gazebo)/models"/>
    <env name="GAZEBO_RESOURCE_PATH"    value="${GAZEBO_RESOURCE_PATH}:$(find rotors_gazebo)/models"/>

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name"          value="$(find rotors_gazebo)/worlds/$(arg world_name).world"/>
        <arg name="debug"               value="$(arg debug)" />
        <arg name="paused"              value="$(arg paused)" />
        <arg name="gui"                 value="$(arg gui)" />
        <arg name="verbose"             value="$(arg verbose)"/>
    </include>

    <group ns="$(arg mav_name)">
        <include file="$(find rotors_gazebo)/launch/spawn_mav.launch">
        <arg name="mav_name"            value="$(arg mav_name)" />
        <arg name="model"               value="$(find auto_landing)/models/mav_with_camera.gazebo" />
        <arg name="enable_logging"      value="$(arg enable_logging)" />
        <arg name="enable_ground_truth" value="$(arg enable_ground_truth)" />
        <arg name="log_file"            value="$(arg log_file)"/>
        </include>
    </group>

    <!-- <node pkg="auto_landing" type="auto_landing_tracking_node" name="auto_landing_tracking_node" output="screen">
			<remap from="command_pose" 	     to="$(arg mav_name)/command/pose" />
			<remap from="husky_odometry"     to="/odometry/filtered" />
			<remap from="quad_odometry"  	 to="$(arg mav_name)/ground_truth/odometry" />
			<remap from="husky_cmd_velocity" to="/husky_velocity_controller/cmd_vel" />
			<remap from="model_state" 		 to="/gazebo/model_states" />
		</node> -->

        <!-- <node pkg="auto_landing" type="auto_landing_husky_node" name="auto_landing_husky_node" output="screen" args="-trajectory $(arg trajectory)">
            <rosparam command="load" file="$(find auto_landing)/cfg/husky_trajectory_params.yaml" />
            <remap from="command_velocity" to="husky_velocity_controller/cmd_vel" />
			<remap from="model_state" 		 to="/gazebo/model_states" />
        </node> -->
</launch>  